File: leetcode/1768.交替合并字符串/main.go
package main

func mergeAlternately(word1 string, word2 string) string {
	word1L, word2L := len(word1), len(word2)

	minL := min(word1L, word2L)

	var newS string

	for i := 0; i < minL; i++ {
		newS += string(word1[i])
		newS += string(word2[i])
	}

	return newS + string(word2[minL:]) + string(word1[minL:])
}

func min(x, y int) int {
	if x < y {
		return x
	}

	return y
}

func mergeAlternately1(word1 string, word2 string) string {
	word1L, word2L := len(word1), len(word2)

	newS := make([]byte, 0, word1L+word2L)

	i, j := 0, 0

	for i < word1L || j < word2L {
		if i < word1L {
			newS = append(newS, word1[i])
			i++
		}

		if j < word2L {
			newS = append(newS, word2[j])
			j++
		}
	}

	return string(newS)
}

File: leetcode/1768.交替合并字符串/main.py
from itertools import zip_longest


class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        word1_l, word2_l = len(word1), len(word2)
        i, j = 0, 0

        new_s = ""

        while i < word1_l or j < word2_l:
            if i < word1_l:
                new_s += word1[i]
                i += 1

            if j < word2_l:
                new_s += word2[j]
                j += 1

        return new_s

    def mergeAlternately1(self, word1: str, word2: str) -> str:
        new_s = ""

        for x, y in zip_longest(word1, word2):
            if x:
                new_s += x
            if y:
                new_s += y

        return new_s
